File: 3010.Divide-an-Array-Into-Subarrays-With-Minimum-Cost-I.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minimumCost = function (nums) {
  const n = nums.length;
  let answer = Infinity;
  let secondMin = nums[1];

  for (let j = 2; j < n; j++) {
    answer = Math.min(answer, nums[0] + secondMin + nums[j]);

    secondMin = Math.min(secondMin, nums[j]);
  }

  return answer;
};

File: python/3010.Divide-an-Array-Into-Subarrays-With-Minimum-Cost-I.py
class Solution:
    def minimumCost(self, nums: List[int]) -> int:
        answer = float("inf")
        second_min = nums[1]

        for i in range(2, len(nums)):
            answer = min(answer, nums[0] + second_min + nums[i])
            second_min = min(second_min, nums[i])

        return answer